### PR TITLE
fix(resource): clean up resource and 100% coverage

### PR DESCRIPTION
**Pull request by mpiercy827**
- [x] Removes some unnecessary parts of code (comments, `$includeMeta` param to `all` method, and excessive exceptions)
- [x] Adds tests for specifying an API version to get up to 100% code coverage

### CODE REVIEW

File: src/Lob/Resource.php [hunk 1]
```diff
@@ -91,57 +78,41 @@ protected function resourceName()
...
             throw new NetworkErrorException($e->getMessage());
-            //@codeCoverageIgnoreEnd
+            // @codeCoverageIgnoreEnd
         } catch (GuzzleException $e) {
             $responseErrorBody = strval($e->getResponse()->getBody());
             $errorMessage = $this->errorMessageFromJsonBody($responseErrorBody);
             $statusCode = $e->getResponse()->getStatusCode();
 
-            if ($statusCode === 401)
+            if ($statusCode === 401 || $statusCode === 403)
```
Comment (dmlittle) [inline src/Lob/Resource.php]: HTTP 403 should be returning a `ForbiddenException`

File: src/Lob/Resource.php [hunk 2]
```diff
@@ -91,57 +78,41 @@ protected function resourceName()
...
-
-            if ($method == 'POST' && $statusCode === 404)
-                throw new ResourceNotFoundException($errorMessage, 404);
-
             if ($statusCode === 422)
                 throw new ValidationException($errorMessage, 422);
 
             // @codeCoverageIgnoreStart
-            // must induce serverside error to test this, so not testable
+            throw new NetworkErrorException($e->getMessage());
```
Comment (dmlittle) [inline src/Lob/Resource.php]: This seems like a copy-pasta extra line

File: src/Lob/Resource.php [hunk 3]
```diff
@@ -170,8 +141,8 @@ protected function getOptions($version, $clientVersion, array $body = null, arra
             $options['headers'] = array_merge($options['headers'], $headers);
         }
 
-        if ($version) {
-            $options['headers']['Lob-Version'] = $version;
+        if ($this->lob->getVersion()) {
```
Comment (dmlittle) [inline src/Lob/Resource.php]: A neat feature of PHP is that you can assign a value of a variable to an expression inside an if condition. When assigning a variable inside an if statement, the expression value is first assigned to the variable and then variable is evaluated as to whether it is truthy. This prevents the second cal to `getVersion()`

```php 
if ($version = $this->lob->getVersion()) {
  $options['headers']['Lob-Version'] = $version;
}
```

File: src/Lob/Resource.php (hunk 2 above)
Comment (mpiercy827) [reply src/Lob/Resource.php]: whoops, nice catch.

File: src/Lob/Resource.php [hunk 4]
```diff
@@ -35,28 +36,19 @@ public function __construct(Lob $lob)
...
     {
-        $all = $this->sendRequest(
+        return $this->sendRequest(
             'GET',
-            $this->lob->getVersion(),
-            $this->lob->getClientVersion(),
             $this->resourceName(),
             $query
         );
-        if ($includeMeta) {
```
[the remark this answers is not in the document]
Comment (mpiercy827) [reply src/Lob/Resource.php]: After doing some digging, I'm not sure if there's a need for it. It was added ~3 years ago, and isn't present in any of the other libraries. I'm planning on releasing a new major version, and am hoping to get a lot of big changes in before that version.

**Pull request merged**